pkg/router/controller/factory/factory.go

```diff
@@ -18,17 +18,48 @@ import (
 	"github.com/openshift/origin/pkg/router/controller"
 )
 
+// RouterControllerFactory initializes and manages the watches that drive a router
+// controller. It supports optional scoping on Namespace, Labels, and Fields of routes.
+// If Namespace is empty, it means "all namespaces".
 type RouterControllerFactory struct {
-	KClient  kclient.Interface
-	OSClient osclient.Interface
+	KClient        kclient.EndpointsNamespacer
+	OSClient       osclient.RoutesNamespacer
+	ResyncInterval time.Duration
+	Namespace      string
+	Labels         labels.Selector
+	Fields         fields.Selector
 }
 
+// NewDefaultRouterControllerFactory initializes a default router controller factory.
+func NewDefaultRouterControllerFactory(oc osclient.RoutesNamespacer, kc kclient.EndpointsNamespacer) *RouterControllerFactory {
+	return &RouterControllerFactory{
+		KClient:        kc,
+		OSClient:       oc,
+		ResyncInterval: 10 * time.Minute,
```

Comment:
Not that I mind (gives me more time to muck around with the haproxy config and test things inside the container) but the earlier one was a couple of minutes.